Non-Interactive Workflows (#2310)
hashicorp/terraform-website: content/cloud-docs/run/cli.mdx

To enable the CLI-driven workflow, you must:

> 1. Run `terraform login` to authenticate with Terraform Cloud. Alternatively, you can [manually configure credentials in the CLI config file](/cli/config/config-file#environment-variable-credentials.

Reviewer: ```suggestion
1. Run `terraform login` to authenticate with Terraform Cloud. Alternatively, you can [manually configure credentials in the CLI config file or through environment variables](/cli/config/config-file#environment-variable-credentials.
```

Author: Okay now that link is a bit too long. I'm going to remove the long in-text link and add a "refer to" statement instead.

> 1. Run `terraform login` to authenticate with Terraform Cloud. Alternatively, you can [manually configure credentials in the CLI config file](/cli/config/config-file#environment-variable-credentials.

Author: ```suggestion
1. Run `terraform login` to authenticate with Terraform Cloud. Alternatively, Alternatively, you can manually configure credentials in the CLI config file or through environment variables. Refer to [CLI Configuration](/cli/config/config-file#environment-variable-credentials) for details.
```

> 1. Run `terraform login` to authenticate with Terraform Cloud. Alternatively, you can [manually configure credentials in the CLI config file](/cli/config/config-file#credentials).

Reviewer: ```suggestion
1. Run `terraform login` to authenticate with Terraform Cloud. Alternatively, you can [manually configure credentials in the CLI config file](/cli/config/config-file#credentials) or set the token using an environment variable.
```

> 1. Add the `cloud` block to your Terraform configuration. You can define its arguments direcly in your configuration file or supply them through for [non-interactive workflows](#non-interactive-workflows). Refer to [Using Terraform Cloud](/cli/cloud) for configuration details.

Reviewer: ```suggestion
1. Add the [`cloud`](https://www.terraform.io/cli/cloud/settings) block to your Terraform configuration. You can define its arguments directly in your configuration file or supply them through environment variables, which can be useful for [non-interactive workflows](#non-interactive-workflows). Refer to [Using Terraform Cloud](/cli/cloud) for configuration details.
```

Reviewer: not sure if we need full path for the link though

Author: We already have this link in the "refer to" text, so I'm going to remove the one you added in the first sentence. Otherwise, I approve!

> 1. Add the `cloud` block to your Terraform configuration. You can define its arguments direcly in your configuration file or supply them through for [non-interactive workflows](#non-interactive-workflows). Refer to [Using Terraform Cloud](/cli/cloud) for configuration details.

Author: ```suggestion
1. Add the `cloud` block to your Terraform configuration. You can define its arguments directly in your configuration file or supply them through environment variables, which can be useful for [non-interactive workflows](#non-interactive-workflows). Refer to [Using Terraform Cloud](/cli/cloud) for configuration details.
```

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

> ### Non-Interactive Workflows

Author: ```suggestion
### Non-Interactive Workflows

> **Hands On:** Try the [Deploy Infrastructure with Terraform Cloud and CircleCI](https://learn.hashicorp.com/tutorials/terraform/circle-ci) tutorial on HashiCorp Learn.
```

External systems cannot run the traditional apply workflow because Terraform requires console input from the user to approve plans. We recommend using the [API-driven Run Workflow](/cloud-docs/run/api) for non-interactive workflows when possible.

> If you prefer to use the CLI, we recommend first running a [speculative plan](/cloud-docs/run#speculative-plans) to preview the changes Terraform will make to your infrastructure. Then, use one of the following approaches based on the [execution mode](/cloud-docs/workspaces/settings#execution-mode) for your workspaces.

Reviewer: ```suggestion
If you prefer to use the CLI in a non-interactive environment, we recommend first running a [speculative plan](/cloud-docs/run#speculative-plans) to preview the changes Terraform will make to your infrastructure. Then, use one of the following approaches based on the [execution mode](/cloud-docs/workspaces/settings#execution-mode) for your workspaces.
```

- **Local Execution:**  Save the approved speculative plan and then run `terraform apply -auto-approve` with the saved plan.

> - **Remote Execution:** Terraform Cloud does not support uploding a saved plan, so we recommend running `terraform apply -auto-approve` immediately after approving the speculative plan to reduce the risk of drift. The [`-auto-approve`](/cli/commands/apply#auto-approve) flag skips the plan approval step in the apply workflow.

Reviewer: ```suggestion
- **Remote Execution:** Terraform Cloud does not support uploading saved plans for remote execution, so we recommend running `terraform apply -auto-approve` **immediately** after approving the speculative plan to prevent the plan from becoming stale. The [`-auto-approve`](/cli/commands/apply#auto-approve) flag skips prompting for approval.
```

> - **Remote Execution:** Terraform Cloud does not support uploading saved plans for remote execution, so we recommend running `terraform apply -auto-approve` **immediately** after approving the speculative plan to prevent the plan from becoming stale. The [`-auto-approve`](/cli/commands/apply#auto-approve) flag skips prompting for approval.

Author: ```suggestion
- **Remote Execution:** Terraform Cloud does not support uploading saved plans for remote execution, so we recommend running `terraform apply -auto-approve` **immediately** after approving the speculative plan to prevent the plan from becoming stale. 
```

>   ~> **Warning:** Remote execution with non-interactive workflows requires auto-approved deployments. Minimize risk of unpredictable infrastructure changes and configuration drift by making sure that no one can change your infrastructure outside of your automated build pipeline. 

Reviewer 2: ```suggestion
  !> **Warning:** Remote execution with non-interactive workflows requires auto-approved deployments. Minimize risk of unpredictable infrastructure changes and configuration drift by making sure that no one can change your infrastructure outside of your automated build pipeline. 
```

>   ~> **Warning:** Remote execution with non-interactive workflows requires auto-approved deployments. Minimize the risk of unpredictable infrastructure changes and configuration drift by making sure that no one can change your infrastructure outside of your automated build pipeline. 

Author: ```suggestion
  !> **Warning:** Remote execution with non-interactive workflows requires auto-approved deployments. Minimize the risk of unpredictable infrastructure changes and configuration drift by making sure that no one can change your infrastructure outside of your automated build pipeline. 
```
